ptp/generate.go

```diff
@@ -228,14 +228,18 @@ func generateToolpath(argv []string) error {
 				}
 			}
 		} else if line.Command == "M106" {
-			if pwm, ok := line.Params["s"]; ok {
-				if err = writer.SetFanSpeed(int(pwm)); err != nil {
-					return err
+			if _, okP10 := line.Params["P10"]; !okP10 {
```

Comment:
This isn't how `Params` works, it'll be that `line.Params["p"]` will return `10, true` if it's a P10. See the check directly below for looking up the S param and the associated PWM value.

I think this should be `if fanIndex, ok := line.Params["p"]; !ok || fanIndex != 10 {`

Reply:
I would also add a comment above explaining what this check is trying to achieve